File: 28-06-2025-Find_Subsequence_Of_Length_K_With_The_Largest_Sum.py
from collections import Counter
from typing import List

class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        # Sort and get top k values
        copy = sorted(nums, reverse=True)
        count = Counter(copy[:k])  # counts top k elements

        ans = []
        for num in nums:
            if count[num] > 0:
                ans.append(num)
                count[num] -= 1
                if len(ans) == k:
                    break
        return ans

File: 28-06-2025-Find_Subsequence_Of_Length_K_With_The_Largest_Sum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k)
    {
        int n=nums.size();
        vector<int> copy{nums.begin(),nums.end()};

        sort(copy.begin(),copy.end(),greater<int>());

        vector<int> ans;
        unordered_map<int,int> count;

        for(int i=0;i<k;++i)
        {
            count[copy[i]]++;
        }


        for(int i=0;i<n;++i)
        {
            if(count[nums[i]]>0)
            {
                ans.push_back(nums[i]);
                count[nums[i]]--;
            }
            if(ans.size()==k)
                break;
        }
        return ans;
        



    }
};